vm/src/protocol/number.rs

```diff
@@ -216,6 +218,97 @@ impl PyNumberMethods {
+    Power,
+    Negative,
+    Positive,
+    Absolute,
+    Boolean,
+    Invert,
+    Lshift,
+    Rshift,
+    And,
+    Xor,
+    Or,
+    Int,
+    Float,
+    InplaceAdd,
+    InplaceSubtract,
+    InplaceMultiply,
+    InplaceRemainder,
+    InplaceDivmod,
+    InplacePower,
+    InplaceLshift,
+    InplaceRshift,
+    InplaceAnd,
+    InplaceXor,
+    InplaceOr,
+    FloorDivide,
+    TrueDivide,
+    InplaceFloorDivide,
+    InplaceTrueDivide,
+    Index,
+    MatrixMultiply,
+    InplaceMatrixMultiply,
+}
+
+impl PyNumberMethodsOffset {
+    pub fn method(&self, methods: &PyNumberMethods, vm: &VirtualMachine) -> PyResult<&BinaryFunc> {
+        use PyNumberMethodsOffset::*;
+        unsafe {
+            match self {
+                // BinaryFunc
+                Add => ptr::addr_of!(methods.add),
+                Subtract => ptr::addr_of!(methods.subtract),
+                Multiply => ptr::addr_of!(methods.multiply),
+                Remainder => ptr::addr_of!(methods.remainder),
+                Divmod => ptr::addr_of!(methods.divmod),
+                Power => ptr::addr_of!(methods.power),
+                Lshift => ptr::addr_of!(methods.lshift),
+                Rshift => ptr::addr_of!(methods.rshift),
+                And => ptr::addr_of!(methods.and),
+                Xor => ptr::addr_of!(methods.xor),
+                Or => ptr::addr_of!(methods.or),
+                InplaceAdd => ptr::addr_of!(methods.inplace_add),
+                InplaceSubtract => ptr::addr_of!(methods.inplace_subtract),
+                InplaceMultiply => ptr::addr_of!(methods.inplace_multiply),
+                InplaceRemainder => ptr::addr_of!(methods.inplace_remainder),
+                InplaceDivmod => ptr::addr_of!(methods.inplace_divmod),
+                InplacePower => ptr::addr_of!(methods.inplace_power),
+                InplaceLshift => ptr::addr_of!(methods.inplace_lshift),
+                InplaceRshift => ptr::addr_of!(methods.inplace_rshift),
+                InplaceAnd => ptr::addr_of!(methods.inplace_and),
+                InplaceXor => ptr::addr_of!(methods.inplace_xor),
+                InplaceOr => ptr::addr_of!(methods.inplace_or),
+                FloorDivide => ptr::addr_of!(methods.floor_divide),
+                TrueDivide => ptr::addr_of!(methods.true_divide),
+                InplaceFloorDivide => ptr::addr_of!(methods.inplace_floor_divide),
+                InplaceTrueDivide => ptr::addr_of!(methods.inplace_true_divide),
+                MatrixMultiply => ptr::addr_of!(methods.matrix_multiply),
+                InplaceMatrixMultiply => ptr::addr_of!(methods.inplace_matrix_multiply),
+                // UnaryFunc
+                Negative => ptr::null(),
+                Positive => ptr::null(),
+                Absolute => ptr::null(),
+                Boolean => ptr::null(),
+                Invert => ptr::null(),
+                Int => ptr::null(),
+                Float => ptr::null(),
+                Index => ptr::null(),
```

Comment:
this is user error and we never can reach here only with python code. panic using `unreachable!` macro will be preferred.